workers/data_refinery_workers/processors/smasher.py

```diff
@@ -478,6 +483,102 @@ def sync_quant_files(output_path, files_sample_tuple, job_context: Dict):
+        "data": None
+    }
+
+    def unsmashable(path):
+      frame['unsmashable_file'] = path
+      return frame
+
+    def smashable(data):
+      frame['data'] = data
+      return frame
+
+    try:
+        # Download the file to a job-specific location so it
+        # won't disappear while we're using it.
+
+        computed_file_path = job_context["work_dir"] + computed_file.filename
+        computed_file_path = computed_file.get_synced_file_path(path=computed_file_path)
+
+        # Bail appropriately if this isn't a real file.
+        if not computed_file_path or not os.path.exists(computed_file_path):
+            logger.error("Smasher received non-existent file path.",
+                computed_file_path=computed_file_path,
+                computed_file=computed_file,
+                dataset=job_context['dataset'],
+                )
+            return unsmashable(computed_file_path) 
+
+        data = _load_and_sanitize_file(computed_file_path)
+
+        if len(data.columns) > 2:
+            # Most of the time, >1 is actually bad, but we also need to support
+            # two-channel samples. I think ultimately those should be given some kind of
+            # special consideration.
+            logger.info("Found a frame with more than 2 columns - this shouldn't happen!",
+                computed_file_path=computed_file_path,
+                computed_file_id=computed_file.id
+                )
+            return unsmashable(computed_file_path) 
+
+        # via https://github.com/AlexsLemonade/refinebio/issues/330:
+        #   aggregating by experiment -> return untransformed output from tximport
+        #   aggregating by species -> log2(x + 1) tximport output
+        if job_context['dataset'].aggregate_by == 'SPECIES' \
+        and computed_file_path.endswith("lengthScaledTPM.tsv"):
+            data = data + 1
+            data = np.log2(data)
+
+        # Detect if this data hasn't been log2 scaled yet.
+        # Ideally done in the NO-OPPER, but sanity check here.
+        if (not computed_file_path.endswith("lengthScaledTPM.tsv")) and (data.max() > 100).any():
+            logger.info("Detected non-log2 microarray data.", file=computed_file)
+            data = np.log2(data)
+
+        # Explicitly title this dataframe
+        try:
+            # Unfortuantely, we can't use this as `title` can cause a collision
+            # data.columns = [computed_file.samples.all()[0].title]
+            # So we use this, which also helps us support the case of missing SampleComputedFileAssociation
+            data.columns = [sample.accession_code]
+        except ValueError as e:
+            # This sample might have multiple channels, or something else.
+            # Don't mess with it.
+            logger.exception("Smasher found multi-channel column (probably) - skipping!",
+                computed_file_path=computed_file_path
+            )
+            return unsmashable(computed_file.filename)
```

Comment:
Shouldn't this be `computed_file_path` here and the other places unsmashable is called?

Reply:
yeah that was there before i updated the function but i think youre right it should be just the filename